src/bytes.rs

```diff
@@ -2,11 +2,13 @@ use {IntoBuf, Buf, BufMut};
 use buf::Iter;
 use debug;
 
-use std::{cmp, fmt, mem, hash, ops, slice, ptr, usize};
-use std::borrow::Borrow;
-use std::io::Cursor;
-use std::sync::atomic::{self, AtomicUsize, AtomicPtr};
-use std::sync::atomic::Ordering::{Relaxed, Acquire, Release, AcqRel};
+use core::{cmp, fmt, mem, hash, ops, slice, ptr, usize};
+use core::borrow::Borrow;
+use core::sync::atomic::{self, AtomicUsize, AtomicPtr};
+use core::sync::atomic::Ordering::{Relaxed, Acquire, Release, AcqRel};
+
+#[cfg(feature = "std")]
```

Comment:
Did this file need to change? It looks like the whole module is only compiled when `std` is enabled?